kafka-starter-testing/src/main/java/org/galatea/kafka/starter/testing/conversion/ConversionUtil.java

```diff
@@ -4,13 +4,15 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.function.BiFunction;
 import java.util.function.Function;
+import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import javafx.util.Pair;
 
 public class ConversionUtil {
 
-  private final Map<Class, List<Pair<Pattern, Function<String, ?>>>> typeConversionMap = new HashMap<>();
+  private final Map<Class, List<Pair<Pattern, BiFunction<String, Matcher, ?>>>> typeConversionMap = new HashMap<>();
```

Comment:
When you find yourself using a BiFunction type on a public method, it's time to introduce a functional interface.
You're likely familiar, but for anyone following along, you can just tack @FunctionalInterface on an interface with a single, two-argument method and replace BiFunction<> with it. As an added bonus, you won't need type arguments for the inputs anymore. (You may want a generic type parameter for the return, though.) You can use lambdas in the calling code per usual.

Maps of Lists of (even JavaFX) Pairs are fine for an internal implementation. Since this class is stateful, I'd suggest renaming from Util to Registry or Service.
Registry is a little more natural, but that implies that you'd be returning a converter rather than doing a conversion; then again, you may find yourself wishing you had put the List<Pair,...> in a PatternConverter, and had a registry that only did Class/fieldPath resolution logic – especially if you end up defining additional types of converters. But this doesn't seem like a high risk.
¯\_(ツ)_/¯

Reply:
fixed